packages/eslint-plugin/src/rules/no-misused-promises.ts

```diff
@@ -369,6 +381,71 @@ export default createRule<Options, MessageId>({
       }
     }
 
+    function checkClassLikeOrInterfaceNode(
+      node:
+        | TSESTree.ClassDeclaration
+        | TSESTree.ClassExpression
+        | TSESTree.TSInterfaceDeclaration,
+    ): void {
+      const tsNode = services.esTreeNodeToTSNodeMap.get(node);
+
+      const heritageTypes = getHeritageTypes(checker, tsNode);
+      if (heritageTypes === undefined || heritageTypes.length === 0) {
```

Comment:
[Style] Nitty nit ✨ 

```suggestion
      if (!heritageTypes?.length) {
```

Reply:
Haha I've gotten mini-traumatized before from a bug related to `!undefined` and `!0` both resolving to `true` (when we just meant to check for `undefined`), so I've tended to make these checks more explicit since then. Which I guess is in line with the [strict-boolean-expressions](https://typescript-eslint.io/rules/strict-boolean-expressions/) TS-Eslint rule? But no strong feelings if you prefer the more concise version, happy to change it :)

Reply:
No strong preference from me either. IMO if it was important we'd get around to writing a lint rule for it. 😄

Reply:
Ok cool committed :)